#! /bin/sh /usr/share/dpatch/dpatch-run
## 52_fix_implicit_declaration.dpatch by Fabrice Coutadeur <fabricesp@ubuntu.com>
##
## Origin: http://lists.opensuse.org/opensuse-commit/2009-07/msg00981.html
## Description: Fix "implicitly converted to pointer at ignoregui.c:192" error.
## This fixes a FTBFS in ia64 and amd64

@DPATCH@
diff -Naur xchat-2.8.6foo/src/fe-gtk/ignoregui.c xchat-2.8.6/src/fe-gtk/ignoregui.c
--- xchat-2.8.6foo/src/fe-gtk/ignoregui.c	2008-02-05 11:02:52.000000000 +0100
+++ xchat-2.8.6/src/fe-gtk/ignoregui.c	2010-03-27 19:10:03.000000000 +0100
@@ -38,6 +38,7 @@
 #include <gtk/gtkliststore.h>
 #include <gtk/gtktreeview.h>
 #include <gtk/gtktreeselection.h>
+#include <gtk/gtkcelllayout.h>
 #include <gtk/gtkcellrenderertext.h>
 #include <gtk/gtkcellrenderertoggle.h>
 
@@ -189,7 +190,7 @@
 	for (col_id=0; (col = gtk_tree_view_get_column (GTK_TREE_VIEW (view), col_id));
 	     col_id++)
 	{
-		GList *list = gtk_tree_view_column_get_cell_renderers (col);
+		GList *list = gtk_cell_layout_get_cells (GTK_CELL_LAYOUT (col));
 		GList *tmp;
 
 		for (tmp = list; tmp; tmp = tmp->next)
